src/BridgedRoom.ts

```diff
@@ -176,6 +176,34 @@ export class BridgedRoom {
         this.puppetOwner = opts.puppet_owner;
 
         this.dirty = true;
+
+        // This almost certainly isn't the best place for this
+        this.joinAllSlackUsers();
+    }
+
+    public async joinAllSlackUsers() {
+        const members = await this.getSlackChannelMembership();
+        if (!members) return;
```

Comment:
```suggestion
```

Don't bother doing this, a `Promise.all([].map(...))` will do the right thing.

Reply:
The typescript compiler disagrees with you.

Reply:
Oh bother, because it can be undefined? Fair